src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,186 @@
+                module,
+                VecBuilder.fill(MODEL_TOLERANCE.get(0, 0)),
+                VecBuilder.fill(SENSOR_TOLERANCE.get(0, 0)),
+                Constants.LOOP_PERIOD
+        );
+        linearSystemLoop = new LinearSystemLoop<N1, N1, N1>(
+                module, quadraticRegulator, kalmanFilter, NOMINAL_VOLTAGE, Constants.LOOP_PERIOD
+        );
+    }
+
+    /**
+     * Sets the angle PID controller for the specific module.
+     *
+     * @param anglePID is the array of PID configurations. [kP, kI, kD, kF]
+     */
+    public void setAnglePID(double[] anglePID) {
+        angleMotor.config_kP(0, anglePID[0]);
+        angleMotor.config_kI(0, anglePID[1]);
+        angleMotor.config_kD(0, anglePID[2]);
+    }
+
+    /**
+     * Sets the drive PID controller for the specific module.
+     *
+     * @param drivePID is the array of PID configurations. [kP, kI, kD, kF]
+     */
+    public void setDrivePID(double[] drivePID) {
+        driveMotor.config_kP(0, drivePID[0]);
+        driveMotor.config_kI(0, drivePID[1]);
+        driveMotor.config_kD(0, drivePID[2]);
+    }
+
+    /**
+     * Sets the motor ports for the specific module.
+     *
+     * @param ports is the array of ports. [drive, angle]
+     */
+    public void setMotorPorts(int[] ports) {
+        driveMotor = new WPI_TalonFX(ports[0]);
+        angleMotor = new WPI_TalonSRX(ports[1]);
+    }
+
+    /**
+     * Configures the inversions for the motors.
+     *
+     * @param inverted is the array of inversions. [drive, angle]
+     */
+    public void configInverted(boolean[] inverted) {
+        driveMotor.setInverted(inverted[0]);
+        angleMotor.setInverted(inverted[1]);
+    }
+
+    /**
+     * Configure sensor phase for module.
+     *
+     * @param sensorPhase is the array of sensor phases. [drive, angle]
+     */
+    public void configSensorPhase(boolean[] sensorPhase) {
+        driveMotor.setSensorPhase(sensorPhase[0]);
+        angleMotor.setSensorPhase(sensorPhase[1]);
+    }
+
+    /**
+     * Sets the postions of the encoders.
+     *
+     * @param positions is the array of positions. [drive, angle]
+     */
+    public void setZeroPosition(int[] positions) {
+        driveMotor.setSelectedSensorPosition(positions[0]);
+        angleMotor.setSelectedSensorPosition(positions[1]);
+    }
+
+    /**
+     * Gets the current angle of the module.
+     *
+     * @return the angle of the module. [deg]
+     */
+    public double getAngle() {
+        return Math.IEEEremainder(
+                angle_unitModel.toUnits(angleMotor.getSelectedSensorPosition()),
+                360
+        );
+    }
+
+    /**
+     * Sets the angle for the module.
+     *
+     * @param reqAngle is the angle to set the module to. [deg]
+     */
+    public void setAngle(double reqAngle) {
+        double currAngle = getAngle();
+
+        currAngle %= 360;
+        currAngle = (currAngle < 0) ? (360 + currAngle) : currAngle;
```

Comment:
This syntax doesn't feel the most natural to me. Consider one of the following suggestions:
```suggestion
        currAngle = (currAngle > 0) ? currAngle : (currAngle + 360);
```
Note I moved the `360` after the variable, feels better to me.
```suggestion
        if (currAngle < 0) {
            currAngle += 360;
        }
```

Reply:
But this line of code runs only if the angle is negative.

Reply:
To me this feels better.

Reply:
> But this line of code runs only if the angle is negative.

The ternary condition runs whether the condition evaluates to truth or not.
I see your point, although because you don't do anything really if `currAngle > 0`, you shouldn't use a ternary operator... You usually use one when you have two different cases, here you have only one.